jina/orchestrate/flow/base.py

```diff
@@ -471,6 +473,12 @@ def _add_gateway(
 
         kwargs.update(self._common_kwargs)
         args = ArgNamespace.kwargs2namespace(kwargs, set_gateway_parser())
+
+        if not (
+            is_port_free(__default_host__, args.port)
```

Comment:
I don't think this is correct. You are checking if the port defined for the Flow on localhost is available. But the Flow will not necessarily run on localhost!
You may build a Flow on one machine, generate K8s configs and deploy them elsewhere. Your could would fail in this case when port is occupied on the building machine, which is wrong.

I think you rather need to check this when the Flow starts. I dont see how you can do this on build time as it is impossible to say where the Flow will be started.

Reply:
Arf, you are completly right:

To add some context: I wanted initially to only check if the port is free before starting the grpc server. However because of some flakyness in the healt check it was not working proprely (there is still this check btw). Testing at parsing time is a hack see this pr for more info about the flakyness/bug : https://github.com/jina-ai/jina/pull/4565

I hack would be to check at parsing time only when run locally. 

The actual true solution is to fix the problem of the pr above, but we decide with @JoanFM that is was not worth it investigate as we move more and more to use k8s

Reply:
The issue with this is that it is basically impossible to know upfront if a Flow will run locally or not? You can't anticipate if the user will eventually start the Flow locally or generate some configs.
I think I dont really get the reasoning why you cant just move the check from build time to start time? If you just have this check on `Flow.start()` it should behave the same no? The only difference is that we can be sure then that the Flow is indeed running locally.

Reply:
yes, it should be moved to `start`